src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            Node<T> node = new Node<>();
+            node.value = value;
+            head = node;
+            tail = node;
+
+        } else {
+            Node<T> node = new Node<>();
+            node.value = value;
+            node.previous = tail;
+            tail.next = node;
+            tail = node;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index is out of bounds");
+        }
+        if (index == size) {
+            Node<T> previousTail = tail;
+            tail = new Node<>();
+            tail.previous = previousTail;
+            tail.value = value;
+            if (previousTail == null) {
+                head = tail;
+            } else {
+                previousTail.next = tail;
+            }
+        } else {
+            Node<T> nodeBeforeInput = getNode(index);
+            Node<T> oldPrevious = nodeBeforeInput.previous;
+            Node<T> inputNode = new Node<>();
+            inputNode.previous = oldPrevious;
+            inputNode.value = value;
+            inputNode.next = nodeBeforeInput;
```

Comment:
pass all three values to Node constructor

Reply:
done